### DISCUSSION TITLE
Counter demo on Python

### DISCUSSION THREAD

**Discussion by laloweb99** [zeroc-ice/ice-demos#141] [General]
Hello! Is it possible to add/do [this c++ example](https://github.com/zeroc-ice/ice-demos/tree/3.7/cpp11/IceStorm/counter) on Python? If is it possible how?

**Comment by pepone** [comment]
Did you check the [python IceStorm demo](https://github.com/zeroc-ice/ice-demos/tree/3.7/python/IceStorm/clock)?

**Comment by laloweb99** [comment]
yes but it does not use any mutex or Ice::uncheckedCast on it :crying_cat_face:

**Comment by pepone** [comment]
The Python hello demo shows the use of ucheckedCast

https://github.com/zeroc-ice/ice-demos/blob/7a91b5461d8358753e766bd1b12cd505d1f093f1/python/Ice/hello/client.py#L20-L23

As per mutex usage, this is a C++ standard API not Ice specific, python has its own APIs for this

see for example:
https://docs.python.org/3/library/threading.html#lock-objects
https://docs.python.org/3/library/threading.html#condition-objects

Some of the demos in https://github.com/zeroc-ice/ice-demos/tree/3.7/python/Ice make use of these python APIs too

**Comment by laloweb99** [comment]
Thank you so much gentleman!
